src/stream/src/executor/mod.rs

```diff
@@ -606,6 +608,22 @@ pub async fn expect_first_barrier(
     Ok(barrier)
 }
 
+/// Expect the first message of the given `stream` as a barrier.
+#[track_caller]
+pub async fn expect_first_barrier_from_aligned_stream(
+    stream: &mut (impl AlignedMessageStream + Unpin),
+) -> StreamExecutorResult<Barrier> {
+    let message = stream
+        .next()
+        .stack_trace("expect_first_barrier")
+        .await
+        .expect("failed to extract the first message: stream closed unexpectedly")?;
+    let barrier = message
+        .into_barrier()
+        .expect("the first message must be a barrier");
+    Ok(barrier)
+}
```

Comment:
add a help function to assert the first init barrier on AlignedMessageStream and used in HashJoin and DynFilter